### ISSUE TITLE
Create query builder like the Typescript client library has

### ISSUE THREAD

**Issue by jerber** [edgedb/edgedb-python#292]
The TS client library has a very cool auto-generated query builder. With python typing we could build the same type of functionality. Here is the TS docs: https://www.edgedb.com/docs/clients/01_js/generation

**Comment by nandorojo** [comment]
Would love to have this for Python.

**Comment by fantix** [comment]
Yes, this is absolutely part of the plan. Let's keep this issue open until it's implemented.

**Comment by Pentusha** [comment]
You may be interested in a [project](https://github.com/Pentusha/edgeql-qb) that I made over a couple of weekends.
The current features set can be best seen in the [file](https://github.com/Pentusha/edgeql-qb/blob/master/tests/test_render.py).
I tried to make it look like SQLAlchemy Core.

**Comment by rmzr7** [comment]
@fantix do you have an estimated timeline for this?

**Comment by fantix** [comment]
> @fantix do you have an estimated timeline for this?

This has been discussed with the team again this week. It's likely that we'll soon come up with something that works in a similar way across Go and Python, so let's sit tight 😼

**Comment by hwmrocker** [comment]
This feature [was just released](https://www.edgedb.com/blog/typesafe-database-querying-via-code-generation), or are you talking about something different?

**Comment by chrisemke** [comment]
is there any news about this? should I use pentusha's project until it officially launches?

**Comment by fantix** [comment]
The codegen will likely be the main way to use for at least the Python binding, this is also demonstrated in the [3.0 launch event](https://youtu.be/bqD5CThzmw4?t=1437).

**Comment by triumph1** [comment]
can we have template for code generation?

## example

below edgeql generates below python code
```edgeql
select 1;
```

```py
# AUTOGENERATED FROM 'app/queries/test.edgeql' WITH:
#     $ edgedb-py


from __future__ import annotations
import edgedb


async def test(
    executor: edgedb.AsyncIOExecutor,
) -> int:
    return await executor.query_single(
        """\
        select 1;\
        """,
    )
```

but this code doesn't pass strict mypy check since query_single is returning `Any`.
I want to use `typing.cast()` to pass mypy check

```py
return cast(int, await executor.query_single(
        """\
        select 1;\
        """,
    ))
```

how do I achieve this? 🤔

**Comment by raddevon** [comment]
Do you mind to create a separate feature request for this, @triumph1? I don't want it to get lost in this query builder request. It's slightly different and might be more feasible. (Don't know that it is but maybe.)

**Comment by triumph1** [comment]
Sure! I'll make an issue and delete above comment. Thank you! 🙇‍♂️

**Comment by triumph1** [comment]
> Do you mind to create a separate feature request for this, @triumph1? I don't want it to get lost in this query builder request. It's slightly different and might be more feasible. (Don't know that it is but maybe.)

https://github.com/edgedb/edgedb-python/issues/463 done! 👍

**Comment by darinkishore** [comment]
hi! is this still planned?
